build_house.rb

```diff
@@ -1,5 +1,25 @@
     end
-end
\ No newline at end of file
+
+    # Generates a random rhyme in the same pattern as the original nursery
+    # rhyme.
+    def random_rhyme
+        characters = ["the house that Jack built", "the malt that lay in", "the rat that ate", "the cat that killed", "the dog that worried", "the cow with the crumpled horn that tossed", "the maiden all forlorn that milked", "the man all tattered and torn that kissed", "the priest all shaven and shorn that married", "the rooster that crowed in the morn that woke", "the farmer sowing his corn that kept", "the horse and the hound and the horn that belonged to"]
+
+        order = characters.shuffle
+        begin_sent = "This is "
+        line = order.pop + ".\n"
+        rhyme_rand = begin_sent + line
+
+        while !order.empty? 
+            line = order.pop + " " + line
+            rhyme_rand += begin_sent + line
+        end
+
+        rhyme_rand
```

Comment:
This is a really nice implementation!  Very readable.

In case it is interesting, I took a stab at this as well, and came up with the following:

```
        lines = order.length.times.map do |index|
            "This is #{ order[0..index].reverse.join(' ') }."
        end
        lines.join("\n")
```